### ISSUE TITLE
Modifier KEY_102ND (86) inverted when setting keyboard layout to neo or neo_qwertz

### ISSUE THREAD

**Issue by EinEtw4s** [hyprwm/Hyprland#5717]
### Hyprland Version

<details> <summary>System/Version info</summary>

```sh
Hyprland, built from branch  at commit fe7b748eb668136dd0558b7c8279bfcd7ab4d759  (props: bump version to 0.39.1).
Date: Tue Apr 16 16:01:03 2024
Tag: v0.39.1, commits: 4460

flags: (if any)


System Information:
System name: Linux
Node name: archdesktop
Release: 6.8.7-arch1-1
Version: #1 SMP PREEMPT_DYNAMIC Wed, 17 Apr 2024 15:20:28 +0000


GPU information: 
12:00.0 VGA compatible controller [0300]: Intel Corporation DG2 [Arc A770] [8086:56a0] (rev 08) (prog-if 00 [VGA controller])
27:00.0 VGA compatible controller [0300]: Advanced Micro Devices, Inc. [AMD/ATI] Ellesmere [Radeon RX 470/480/570/570X/580/580X/590] [1002:67df] (rev e7) (prog-if 00 [VGA controller])


os-release: NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://gitlab.archlinux.org/groups/archlinux/-/issues"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo


plugins:
```
</summary>
</details>

### Bug or Regression?

Bug

### Description

The modifier key KEY_102ND (86) is inverted should neo or neo_qwertz be selected as the keyboard layout, which is in effect putting the user onto layer 4 by default. The normal default layer without any modifiers being pressed is layer 1.
I confirmed this by looking at `libinput debug-events`. This key tends to be mapped as `<,>,|` on a QWERTZ keyboard by default.

### How to reproduce

Set the following settings in hyprland.conf:
```
kb_layout = de
kb_variant = neo_qwertz
```
or alternatively
```
kb_layout = de
kb_variant = neo
```

Then try to type something.

**Comment by vaxerski** [comment]
that's handled by xkb not hyprland

**Comment by EinEtw4s** [comment]
Thank you, that is totally true. I suspected Hyprland at first, since this issue did not appear with the same settings in the login shell, iirc.

I did also find the solution to my issue in the neo FAQ (https://www.neo-layout.org/Probleme/FAQ/#linux-unix-bsd), should anyone experience the same problem:
Should you be locked in layer 4, the lock can be disengaged by pressing the left and right modifiers for layer 4 (`<,>,|` + `AltGr`) or alternatively `Shift`+`CapsLock/M3L`+`Tab` at the same time.
